### PR TITLE
Autogenerate the a dictionary of translators to hide class inheritance inconsistancy.

### PR DESCRIPTION
**Pull request by Vincent-Stragier**
#129

### CODE REVIEW

**Comment by nidhaloff** [comment]
@2010019970909 can you actually see my comment review? this is a question because I'm not sure

File: deep_translator/deepl.py [hunk 1]
```diff
@@ -83,6 +87,9 @@ def _map_language_to_code(self, lang, **kwargs):
         raise LanguageNotSupportedException(lang)
 
 
+BaseTranslator.register(DeepL)
```
[the remark this answers is not in the document]
Comment (2010019970909) [reply deep_translator/deepl.py:90]: It comes from the `ABC` class (https://docs.python.org/3/library/abc.html#abc.ABCMeta.register):

In [`parent.py`](https://github.com/2010019970909/deep-translator/blob/697d292bb9980536731a9f79d6513740cf0b7e3d/deep_translator/parent.py#L4-L8), you defined:
```python
class BaseTranslator(ABC):
    ...
```

The `register` method will only affect the `issubclass` and `isinstance` functions (according to this [answer](https://stackoverflow.com/a/3392406/10949679)).

**Comment by nidhaloff** [comment]
@Vincent-Stragier I like the idea since it makes the translator_dict unnecessary and the cli implementation more dynamic, so I will merge your PR ;)

I would however prefer to keep the __init__ file clean. Maybe there is a way to move the declaration of __engines__ to another file.

**Comment by Vincent-Stragier** [comment]
@nidhaloff, I totally understand for the init file, but I do not see an elegant solution which do not lead to a recursive import loop.

**Pull request merged**